mindsdb/integrations/handlers/qdrant_handler/qdrant_handler.py

```diff
@@ -0,0 +1,476 @@
+
+        Raises:
+            Exception: If an unsupported operator is specified
+
+        Returns:
+            dict: A dict of Qdrant filtering clauses
+        """
+        if operator == FilterOperator.EQUAL:
+            return {"match": models.MatchValue(value=value)}
+        elif operator == FilterOperator.NOT_EQUAL:
+            return {"match": models.MatchExcept(**{"except": [value]})}
+        elif operator == FilterOperator.LESS_THAN:
+            return {"range": models.Range(lt=value)}
+        elif operator == FilterOperator.LESS_THAN_OR_EQUAL:
+            return {"range": models.Range(lte=value)}
+        elif operator == FilterOperator.GREATER_THAN:
+            return {"range": models.Range(gt=value)}
+        elif operator == FilterOperator.GREATER_THAN_OR_EQUAL:
+            return {"range": models.Range(gte=value)}
+        else:
+            raise Exception(f"Operator {operator} is not supported by Qdrant!")
+
+    def _translate_filter_conditions(
+        self, conditions: List[FilterCondition]
+    ) -> Optional[dict]:
+        """
+        Translate a list of FilterCondition objects a dict that can be used by Qdrant.
+        Filtering clause docs can be found here: https://qdrant.tech/documentation/concepts/filtering/
+        E.g.,
+        [
+            FilterCondition(
+                column="metadata.created_at",
+                op=FilterOperator.LESS_THAN,
+                value=7132423,
+            ),
+            FilterCondition(
+                column="metadata.created_at",
+                op=FilterOperator.GREATER_THAN,
+                value=2323432,
+            )
+        ]
+        -->
+        models.Filter(
+        must=[
+            models.FieldCondition(
+                key="created_at",
+                match=models.Range(lt=7132423),
+            ),
+            models.FieldCondition(
+                key="created_at",
+                match=models.Range(gt=2323432),
+            ),
+          ]
+        )
+        """
+        # We ignore all non-metadata conditions
+        if conditions is None:
+            return None
+        filter_conditions = [
+            condition
+            for condition in conditions
+            if condition.column.startswith(TableField.METADATA.value)
+        ]
+        if len(filter_conditions) == 0:
+            return None
+
+        qdrant_filters = []
+        for condition in filter_conditions:
+            payload_key = condition.column.split(".")[-1]
+            qdrant_filters.append(
+                models.FieldCondition(key=payload_key, **self._get_qdrant_filter(condition.op, condition.value))
+            )
+
+        return models.Filter(must=qdrant_filters) if qdrant_filters else None
+
+    def update(
+        self, table_name: str, data: pd.DataFrame, columns: List[str] = None
+    ) -> HandlerResponse:
+        """
+        Update data in the Qdrant database.
+        TODO: Update for vector DBs has not been implemented.
```

Comment:
please create an issue so we can not forget about supporting this in future

Reply:
#7864.